source/gems/dry-transaction/basic-usage.html.md

```diff
@@ -3,42 +3,111 @@ title: Basic usage
 
-All you need to use dry-transaction is a container to hold your application’s operations. Each operation must respond to `#call(input)`.
+You can define a standalone transaction based around a single class and its own instance methods. Each instance method must accept an input argument and return an output wrapped in a `Right` (for success) or `Left` (for failure):
 
-The operations will be resolved from the container via `#[]`. For our examples, we’ll use a [`dry-container`](http://dry-rb.org/gems/dry-container):
+```ruby
+require "dry/transaction"
+
+class CreateUser
+  include Dry::Transaction
+
+  step :process
+  step :validate
+  step :persist
+
+  def process(input)
+    Right(name: input["name"], email: input["email"])
+  end
+
+  def validate(input)
+    if input[:email].nil?
+      Left(:not_valid)
+    else
+      Right(input)
+    end
+  end
+
+  def persist(input)
+    DB << input
+
+    Right(input)
+  end
+end
+```
+
+### Defining a transaction with an operations container
+
+You can also define a transaction that relies upon operation objects in a container. Each operation must respond to `#call(input)`.
+
+The operations will be resolved from the container via `#[]`. For this example, we’ll use [dry-container](/gems/dry-container):
 
 ```ruby
 require "dry-container"
-require "dry-monads"
+require "dry/transaction"
+require "dry/transaction/operation"
+
+class Process
+  include Dry::Transaction::Operation
+
+  def call(input)
+    Right(name: input["name"], email: input["email"])
+  end
+end
+
+class Validate
+  include Dry::Transaction::Operation
+
+  def call(input)
+    if input[:email].nil?
+      Left(:not_valid)
+    else
+      Right(input)
+    end
+  end
+end
+
+class Persist
+  include Dry::Transaction::Operation
+
+  def call(input)
+    DB << input
+
+    Right(input)
+  end
+end
 
 class Container
   extend Dry::Container::Mixin
 
-  register :process, -> input {
-    Dry::Monads.Right(name: input["name"], email: input["email"])
-  }
+  namespace "operations" do |ops|
+    ops.register "process" do
+      Process.new
+    end
 
-  register :validate, -> input {
-    input[:email].nil? ? Dry::Monads.Left(:not_valid) : Dry::Monads.Right(input)
-  }
+    ops.register "validate" do
+      Validate.new
+    end
 
-  register :persist, -> input {
-    DB << input; Dry::Monads.Right(input)
-  }
+    ops.register "persist" do
+      Persist.new
+    end
+  end
 end
 ```
 
-### Defining a transaction
+n.b. this is a small, contrived container setup. In a real app, you should consider using [dry-system](/gems/dry-system) to make it easier to populate a container with your own objects.
 
-Define a transaction to bring your operations together:
+Once you have a container, you can pass it to your transaction mixin and refer to the registered operations.
 
 ```ruby
-save_user = Dry.Transaction(container: Container) do
-  step :process
-  step :validate
-  step :persist
+class CreateUser
+  include Dry::Transaction
```

Comment:
@tim are we not missing the container in this example 
`include Dry::Transaction(container: Container)`